rustic-rustdoc.el

```diff
@@ -50,6 +49,20 @@ All projects and std by default, otherwise last open project and std.")
                              ()
                              ,(concat rustdoc-source-repo "filter.lua"))))
 
+(defvar rustdoc-search-function (if (and  (require 'helm-ag nil t) (executable-find "rg")) ; If helm-ag and rg is available we use them by default, otherwise revert to using grep
+                                       (lambda (search-dir search-term)
+                                         (let* ((helm-ag-base-command (if rustdoc-current-project ; If the user has not visited a project the search will be done from the doc root, in which case we should not follow symlinks.
+                                                                          "rg -L --smart-case --no-heading --color=never --line-number --pcre2"
+                                                                        "rg --smart-case --no-heading --color=never --line-number --pcre2"))
+                                                (helm-ag-fuzzy-match t)
+                                                (helm-ag-success-exit-status '(0 2)))
```

Comment:
Is this really necessary for this functionality ? These options should generally be chosen by the users.

Reply:
The first one, no. I believe that is a leftover from me playing around with the settings. The second one is from the documentation of helm-ag: "Ripgrep uses exit-status 2 to indicate a partial success:

(custom-set-variables
 '(helm-ag-base-command "rg --no-heading")
 `(helm-ag-success-exit-status '(0 2)))"

I did not set the exit status at first but ran into some surprising behavior which I can't remember now, and subsequently added it.